### PR TITLE
Clearing /tmp on init and finalize

### PR DESCRIPTION
**Pull request by justyoyo123**
## Description
This adds a new function called `clear_tmp()` which is called in `lindrustinit` and `lindrustfinalize` to ensure that we are working with a fresh /tmp after init.

Fixes # (issue)

 

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?
Wrote a new test called `ut_lind_fs_tmp_file_test()` in `fs_tests.rs` to verify behavior of `clear_tmp()` function.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Any dependent changes have been added to a pull request and/or merged in other modules (native-client, lind-glibc, lind-project)

### CODE REVIEW

File: src/safeposix/dispatcher.rs [hunk 1]
```diff
@@ -550,15 +553,33 @@ pub extern "C" fn lindthreadremove(cageid: u64, pthreadid: u64) {
     cage.thread_table.remove(&pthreadid);
 }
 
+fn cleartmp(flag : bool) {
+    let path = "tmp";
```
Comment (RusherRG) [inline src/safeposix/dispatcher.rs:557]:
```suggestion
    let path = "/tmp";
```

To be specific.

File: src/safeposix/dispatcher.rs [hunk 2]
```diff
@@ -550,15 +553,33 @@ pub extern "C" fn lindthreadremove(cageid: u64, pthreadid: u64) {
     cage.thread_table.remove(&pthreadid);
 }
 
+fn cleartmp(flag : bool) {
```
Comment (RusherRG) [inline src/safeposix/dispatcher.rs:556]:
```suggestion
fn cleartmp(init : bool) {
```

File: src/safeposix/dispatcher.rs [hunk 3]
```diff
@@ -550,15 +553,33 @@ pub extern "C" fn lindthreadremove(cageid: u64, pthreadid: u64) {
...
+    let mut statdata = StatData::default();
+    
+    if cage.stat_syscall(path, &mut statdata) == 0 {
+        lib_fs_utils::visit_children(&cage, path, None, |childcage, childpath, isdir, _| {
+            if isdir { lib_fs_utils::lind_deltree(childcage, childpath); }
+            else { childcage.unlink_syscall(childpath);}
+        });
+    }
+    else {
+        if flag == true {
```
Comment (RusherRG) [inline src/safeposix/dispatcher.rs:569]:
```suggestion
        if init == true {
```

File: src/safeposix/dispatcher.rs [hunk 4]
```diff
@@ -595,18 +617,24 @@ pub extern "C" fn lindrustinit(verbosity: isize) {
...
+    cleartmp(init);
 }
 
 #[no_mangle]
 pub extern "C" fn lindrustfinalize() {
-
-    interface::cagetable_clear();
-
+    
+    let path = "tmp";
```
Comment (RusherRG) [inline src/safeposix/dispatcher.rs:628]:
```suggestion

```

File: src/safeposix/dispatcher.rs [hunk 5]
```diff
@@ -595,18 +617,24 @@ pub extern "C" fn lindrustinit(verbosity: isize) {
         sem_table: interface::RustHashMap::new(),
     };
     interface::cagetable_insert(1, initcage);
+    // make sure /tmp is clean
+    let init:bool = true;
+    cleartmp(init);
```
Comment (RusherRG) [inline src/safeposix/dispatcher.rs:622]:
```suggestion
    cleartmp(true);
```
Do we need to create a new variable?

**Review by RusherRG — CHANGES_REQUESTED** [review]
This is a very nice fix and much needed. It will help us a lot while testing out changes without having to clear the filesystem after each test run. Great!

File: src/safeposix/dispatcher.rs [hunk 6]
```diff
@@ -110,6 +110,9 @@ use super::net::{NET_METADATA};
 use crate::interface::errnos::*;
 use super::syscalls::sys_constants::*;
 use super::syscalls::fs_constants::IPC_STAT;
+use crate::lib_fs_utils;
```
Comment (rennergade) [inline src/safeposix/dispatcher.rs:113]: these imports actually should be cleaned up, can you make this just one line?

**Review by rennergade — APPROVED** [review]
looks good, approved!

**Review by RusherRG — APPROVED** [review]

**Pull request merged**